warehouse/query-core/src/main/java/org/apache/commons/jexl2/parser/ASTDelayedPredicate.java

```diff
@@ -44,10 +44,18 @@ public String toString() {
     }
     
     /**
+     * Wrap the provided node in an ASTDelayedPredicate
+     * 
      * @param node
-     * @return
+     *            the node to delay
+     * @return an ASTDelayedPredicate
      */
-    public static ASTDelayedPredicate create(JexlNode node) {
+    public static JexlNode create(JexlNode node) {
+        
+        // Do not delay and already-delayed sub tree
+        if (node instanceof ASTDelayedPredicate || ASTDelayedPredicate.isSubTreeAlreadyDelayed(node)) {
```

Comment:
Can't we use instanceOf(node, ASTDelayedPredicate) here instead?

Reply:
Can do!